### ISSUE TITLE
Not all entities of a Circle seem to have valid values yet.

### ISSUE THREAD

**Issue by CoHoAppel** [plugwise/plugwise-beta#157]
My configuration: USB Stick, Circle+ and 6 more Circles. Latest version 0.14.4 installed. The configuration runs stable and a number of entities (switching on/off; current power usage; power consumption current hour;) work well, however the values of power consumption previous hour; power consumption today /yesterday; on the overview screen remain empty. Are these still work in progress? I checked the Plugwise part by moving the the stick to a Windows PC and "reading" the history with the Plugwise Source application. All connected Circles reported the power-history of the past couple of weeks.

**Comment by bouwew** [comment]
In my system I have the following sensors:
- sensor.power_consumption_current_hour
- sensor.power_consumption_previous_hour
- sensor.power_consumption_today
- sensor.power_consumption_yesterday
- sensor.power_usage
- switch.relay_state

Can you please send a picture of the entities that are shown for one of the Circles?

**Comment by CoHoAppel** [comment]
This is what I see. The enclosed document (comments in Dutch) show screen-shots and a short description of what is going on...

**Comment by bouwew** [comment]
@CoHoAppel Please correspond directly in the chat of the Issue, I don't see anything other than the text you write.

**Comment by CoHoAppel** [comment]
Enclosed the requested pictures with comments.
[3 entities with value zero.docx](https://github.com/plugwise/plugwise-beta/files/6090645/3.entities.with.value.zero.docx)

**Comment by bouwew** [comment]
Thanks for providing the detailed info.
@brefra please have a look.

**Comment by CoHoAppel** [comment]
Updated to the latest version (just in case) and looked at some additional material. Looking at the History tab I notices that the values may be available, but only during a very short moment as they seem to be overwritten by a zero value right after the set-moment. In the History graph this translates into a small spike, which may never show up in the Overview Screen. Screen copies are enclosed (here this time ...)
[Values seem to be there a short moment... and then they are reset again.docx](https://github.com/plugwise/plugwise-beta/files/6097955/Values.seem.to.be.there.a.short.moment.and.then.they.are.reset.again.docx)

**Comment by tvwerkhoven** [comment]
I have a similar issue, also using USB Stick, Circle+ and 3 more Circles in my case. Some entities work OK, others don't report data. So hardware is OK, connection is OK, configuration is OK, but something in de details of populating the entities is going wrong. Any idea what's going on? I have similar problems with multiple Circles.

E.g. see this sensor's entity history:
* sensor.power_usage (Works OK -- entity renamed for convenience)
<img width="399" alt="sensor.power_usage" src="https://user-images.githubusercontent.com/78742/121016720-c478aa80-c79c-11eb-8d89-2415633a0467.png">
* sensor.power_consumption_current_hour (Works OK):
<img width="398" alt="sensor.power_consumption_current_hour" src="https://user-images.githubusercontent.com/78742/121016837-e6722d00-c79c-11eb-92b2-0ca00a81218a.png">
* sensor.power_consumption_previous_hour (not OK - should show same data as previous graph 1 hour time-shifted):
<img width="399" alt="sensor.power_consumption_previous_hour" src="https://user-images.githubusercontent.com/78742/121016886-f853d000-c79c-11eb-979c-d1f87d111c76.png">
* sensor.power_consumption_today (not OK - should show data around the time power_usage was >0 -- entity renamed for convenience)
<img width="396" alt="sensor.power_consumption_today" src="https://user-images.githubusercontent.com/78742/121016941-086baf80-c79d-11eb-9f19-dded807c060d.png">
* sensor.power_consumption_yesterday (works OK)
<img width="396" alt="sensor.power_consumption_yesterday" src="https://user-images.githubusercontent.com/78742/121017018-18838f00-c79d-11eb-8fcd-9dd4091d4ea0.png">

**Comment by bouwew** [comment]
@CoHoAppel 
Please try plugwise-beta v0.17.0a7, do the changes in the Stick-code have any effect?

**Comment by brefra** [comment]
@tvwerkhoven above applies for you as well

**Comment by tvwerkhoven** [comment]
Thanks for the update! I'm now trying v0.18.0, sorry for the wait. Will keep you posted what happens.

**Comment by tvwerkhoven** [comment]
Running for a few days, 'Power usage' and 'Power consumption current hour' are (still) working reliably, but the other power consumptions are not (i.e. 'Power consumption {last hour,today,yesterday}'). Instead of giving a 0 value however, the now report 'unknown'.

I only want to know total (cumulative) power consumption. Could this be added? The other counters are funny/less useful anyway. Thanks! :)

**Comment by bouwew** [comment]
As far as we know, there is no total cumulative value available from the Circles.
But you can create such a sensor yourself, using the Power Usage values, via de (Riemann) Integration:
```
sensor:
  - platform: integration
    source: sensor.power_usage_c2b30
    name: mediacenter_cumulative_energy
    method: left
    unit_prefix: k
    unit_time: h
```

**Comment by bouwew** [comment]
@tvwerkhoven
One of my sensors is showing this:
![image](https://user-images.githubusercontent.com/11290930/135887130-7ce4fd0b-b008-428b-a346-ecc467377330.png)
Working as intended.

Your Stick might be buggy. See earlier messages in this topic. Did you maybe run a beta-version when using Source?  Or when using the Stretch?
The thing is, our code just reads data from the Circle's etc. via the Stick. Either the data the code receives is wrong, or we read from the wrong data register. We are pretty sure we read from the correct register but we're not 100% sure.
I've been trying to get some input directly from Plugwise but it doesn't look like that will ever happen :(

**Comment by tvwerkhoven** [comment]
@bouwew : thanks for the quick reply. What specific earlier message do you mean? I bought these Circles from someone else, not sure what happened before them. I'm only using them with a Stick.

Do your 'Power consumption current day' and other metrics also work as intended? If so, how do they compare to using the Integration approach you propose? If they're close enough that would be a suitable work around. Only drawback is the Integration plugin cannot be assigned to rooms/icons etc.

**Comment by bouwew** [comment]
I am referring to the message by CoHoAppel. He has reported similar and different problems.
No, current today and some others don't seem correct. But I seem to suffer from network issues.

**Comment by brefra** [comment]
@tvwerkhoven, This is a strange issue. At first glance it looks like the internal clock or memory of the Circle is broken/misfunction, which causes the reset of the power consumption counter.
To really understand and debug the issue, a few questions:

- What does the `sensor.energy_consumption_today_xxxxx` show?
- Could you lookup the device info (type and firmware) shown at the device information tab of the device?
- Would you be able set the logging to the "info" level for like half an hour and collect the home-assistant.log file?
You can set the log level using the service call like below
![image](https://user-images.githubusercontent.com/6314181/136095013-14b5bab0-934b-4120-a619-29d1e1e56826.png)
and collect the home-assistant.log file?

**Comment by brefra** [comment]
@CamellaToe you're correct. The circles stores every hour the power used that hour. So indeed it's possible to create a sensor in HA to show the used energy of a particular timeframe.

Actually that's the way i.e. the "Energy Consumption Today" and "Power Consumption Yesterday" (this last one should actually be renamed to 'Energy Consumption Yesterday" but that's a "breaking change")

Is there any specific timeframe you're interested in, to be added as a sensor in HA? 
I'm interested in the use case, as currently the "Energy Consumption Today" sensor supports the statistics function so history energy consumption is already available at HA (from the moment the circle is added to HA)

I know the HA team wants to add functionality to add historical measurements to the statistics database. Collecting all the historical energy use of the circles and import them into HA would be a nice feature for first time users.

**Comment by CamellaToe** [comment]
@brefra I just updated from v0.18.0 to v0.18.1a0 and now the  Power Consumption Yesterday is available.
As far as I remember I have not seen that before so indeed a big step forward.

Having a single measurement per day for solar panels is not enough. The measurements have to be hourly.
The previous hour sounds great. Surely we can get that stored in HA somehow for a nice graph

I'm pretty new to HA but having to read data with a timestamp sounds like a nightmare

**Comment by bouwew** [comment]
Nice find! :+1:

**Comment by tvwerkhoven** [comment]
Thanks @brefra, sounds great! Again, it used to work before, not sure what changed (and not sure if that's helpful).

I've updated the log file with more/different plug activity (4 hours, 480MB - Plugwise is a chatty plugin!): https://nextcloud.vanwerkhoven.org/index.php/s/wft2ms55YxpCjCC
